execution.go

```diff
@@ -0,0 +1,50 @@
+package execution
+
+import (
+	"time"
+
+	execution "github.com/rollkit/go-execution"
+	rollkitTypes "github.com/rollkit/rollkit/types"
+)
+
+type EngineAPIExecutionClient struct {
+	execute execution.Execute
```

Comment:
I don't understand this "redirection" - it's just nested use of the interface, with no extra functionality.

Why you added the extra level of abstraction/nesting?

Reply:
Agreed. This was me trying to design with abstraction during the prototyping phase, which very much is an anti-pattern. I've removed the redundant abstraction.